django/contrib/postgres/fields/array.py

```diff
@@ -288,3 +289,13 @@ def __init__(self, start, end):
 
     def __call__(self, *args, **kwargs):
         return SliceTransform(self.start, self.end, *args, **kwargs)
+
+
+@CombinedExpression.dispatch(CombinedExpression.ADD, ArrayField, None)
+@CombinedExpression.dispatch(CombinedExpression.ADD, None, ArrayField)
+@CombinedExpression.dispatch(CombinedExpression.ADD, ArrayField, ArrayField)
```

Comment:
Do we need the third case, isn't ArrayField, None enough to cover ArrayField, ArrayField, too.

Field would read much better than None for the catch-all case.

What happens if you have also a rule CharField, Field -> StringConcat, and you run a query with F('char_field') + F('array_field')?